tests/jobs/test_local_task_job.py

```diff
@@ -544,10 +535,53 @@ def test_process_kill_calls_on_failure_callback(self, signal_type, dag_maker):
         def failure_callback(context):
             with shared_mem_lock:
                 failure_callback_called.value += 1
-            assert context['dag_run'].dag_id == 'test_mark_failure'
+            assert context['dag_run'].dag_id == 'test_send_sigkill'
 
         def task_function(ti):
+            # pylint: disable=unused-argument
+            time.sleep(1)
+            os.kill(ti.pid, signal.SIGKILL)
```

Comment:
Let's be explicit --we are killing ourselves here
```suggestion
            os.kill(os.getpid(), signal.SIGKILL)
```